servers/physics_2d/body_pair_2d_sw.cpp

```diff
@@ -403,12 +403,12 @@ bool BodyPair2DSW::pre_solve(real_t p_step) {
 		c.rA = global_A;
 		c.rB = global_B - offset_B;
 
-		if (A->can_report_contacts()) {
+		if (A->can_report_contacts() && B->layer_in_mask(A)) {
```

Comment:
It would be nice to have a member or at least local variable to avoid repeating the same logic multiple times.

Reply:
I prefer it this way. It's the same format as the other condition, it's method name makes it easy to understand when reading, and the compiler will inline it anyway.